device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -701,14 +701,24 @@ public void onAuthenticationFailed(String deviceId, TransportException transport
     {
         if (this.isMultiplexing)
         {
-            if (this.savedException == null)
+            if (this.state != IotHubConnectionStatus.CONNECTED)
             {
-                this.savedException = new MultiplexingDeviceUnauthorizedException("One or more multiplexed devices failed to authenticate");
-            }
+                if (this.savedException == null)
+                {
+                    this.savedException = new MultiplexingDeviceUnauthorizedException("One or more multiplexed devices failed to authenticate");
```

Comment:
It might be a naive question, but do we distinguish between Authentication and Authorization?

Reply:
On AuthenticationFailed we are setting the savedException as an Unauthorized exception.

Reply:
I use them interchangeably, so there is no difference as far as I'm aware